src/shared/calendarspec.c

```diff
@@ -1196,12 +1197,19 @@ static int tm_within_bounds(struct tm *tm, bool utc) {
                 return negative_errno();
 
         /* Did any normalization take place? If so, it was out of bounds before */
-        int cmp = CMP(t.tm_year, tm->tm_year) ?:
-                  CMP(t.tm_mon, tm->tm_mon) ?:
-                  CMP(t.tm_mday, tm->tm_mday) ?:
-                  CMP(t.tm_hour, tm->tm_hour) ?:
-                  CMP(t.tm_min, tm->tm_min) ?:
-                  CMP(t.tm_sec, tm->tm_sec);
+        if ((cmp = CMP(t.tm_year, tm->tm_year)) != 0)
+                t.tm_mon = 0;
```

Comment:
shouldn't this be 1?

Reply:
mktime(3) says valid tm_mon is 0-11, so I think tm_mon should be 0. https://github.com/systemd/systemd/blob/0be1504704caff1c48618d100c5d4ae0b4198485/src/shared/calendarspec.c#L1272-L1274
It looks like systemd internally uses +1 biased month, but restored after use.